pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -668,17 +789,30 @@ func (nc *Controller) doEvictionPass() {
 				klog.Warningf("Failed to get Node %v from the nodeLister: %v", value.Value, err)
 			}
 			nodeUID, _ := value.UID.(string)
+
+			// TODO: pods on nodes are for all TPs, need an efficient way to get subset of pods for the TP to the DeletePods function
 			for _, tpManager := range nc.tenantPartitionManagers {
-				remaining, err := nodeutil.DeletePods(tpManager.Client, nc.recorder, value.Value, nodeUID, tpManager.DaemonSetStore)
+				pods, err := tpManager.PodByNodeNameLister(value.Value)
+
 				if err != nil {
+					utilruntime.HandleError(fmt.Errorf("unable to list pods from node %q: %v", value.Value, err))
```

Comment:
If tpManager is TP1, and node is from TP2, what will the output? Could it be some error?

Reply:
the node is the same for all TPs. with multiple TPs, as long as one TP failed and the function retuns false to the TRY function which will replace the workitem in the queue and redo in the next iteration till ALL pods on all TPs were evicted for the nodes.

Reply:
there is a potential case here where IF, TP1 keeps failing and it will affect the eviction pods from other TPs. but this can be fixed later on ( i guess use continue instead of break in the inner loop or run eviction in TP in parallel can solve this.

Reply:
fixed

Reply:
Upon discussion, please make sure UT check whether error is possible if there is no pods existed for this TP and this node. Also can continue if len(pods) == 0.

Reply:
fixed and added check for empty length of pods from one TP